### PR TITLE
Support mode w/o quotes in the transport header and made it case-insensitive

### LINKED ISSUE #18: Transport header's parsing
Hello! Great work, I'm glad it exists and I can use it. Thanks!
But unfortunately I've got trouble where we couldn't specify some transport headers. Well, if _mode_'s value isn't wrapped with quotes, we'll be getting error all the time. But RFC 2326 declares absence of quotes as an alternative to quoted variant, so they're kind of optional. 

Here it was declared: [RFC 2326](https://www.rfc-editor.org/rfc/rfc2326), page 60

### PR DESCRIPTION
**Pull request by r4v3n6101**
Close #18

### CODE REVIEW

**Comment by r4v3n6101** [comment]
Maybe there's a need to bump version of the crate, but I'm not sure

File: src/headers/transport.rs [hunk 1]
```diff
@@ -364,8 +362,8 @@ pub enum TransportMode {
 impl<'a> From<&'a str> for TransportMode {
     fn from(transport_mode: &'a str) -> TransportMode {
         match transport_mode {
-            "PLAY" => TransportMode::Play,
-            "RECORD" => TransportMode::Record,
+            "PLAY" | "play" => TransportMode::Play,
+            "RECORD" | "record" => TransportMode::Record,
```
[the remark this answers is not in the document]
Comment (sdroege) [reply src/headers/transport.rs:366]: Also RFC 7826 Section 4.2:

> All other parts [apart from scheme/host] of RTSP URIs and IRIs are case sensitive, and they MUST NOT be case mapped.

**Comment by sdroege** [comment]
> Maybe there's a need to bump version of the crate, but I'm not sure

That's fine, I'll do that when releasing.

The clippy failure should be gone now if you rebase on latest main.

File: src/headers/transport.rs (hunk 1 above)
[the remark this answers is not in the document]
Comment (r4v3n6101) [reply src/headers/transport.rs:366]: Yes, I do have such server. But it seems slightly non-standard RTSP. I may remove lower-case matches according to RFC. Should I?

[the remark this answers is not in the document]
Comment (sdroege) [reply src/headers/transport.rs:366]: If you actually have such a server then that's probably useful to handle, but please add a comment to the code about that.

OOC, what server is that?

[the remark this answers is not in the document]
Comment (r4v3n6101) [reply src/headers/transport.rs:366]: Apple RAOP protocol (RTSP part of its). It consists of `Transport` header contains `mode=play` or `mode=record`.

[the remark this answers is not in the document]
Comment (sdroege) [reply src/headers/transport.rs:366]: Yeah they're not going to be open to change that. Let's support that then :)

It's only a problem when parsing the responses from it, it doesn't have any problems with the requests we produce?

[the remark this answers is not in the document]
Comment (r4v3n6101) [reply src/headers/transport.rs:366]: I haven't seen such problem. Maybe later I'll get it, but for now these changes are enough.

**Review by sdroege — APPROVED** [review]

**Comment by sdroege** [comment]
Do you need a release with this or do you first want to see if you need more fixes for RAOP?

**Comment by r4v3n6101** [comment]
That's fine I think. Maybe bump a version to catch latest changes via crates.io?

**Pull request merged**